### ISSUE TITLE
Fleche Improvements

### ISSUE THREAD

**Issue by Iridar** [long-war-2/lwotc#1311]
Fleche uses unit's starting position to calculate its bonus damage, which may not necessarily correspond to unit's travel distance while using Fleche. 

I believe this `AbilityContext.InputContext.MovementPaths[0].MovementTiles.Length` will return the distance traveled in tiles, which should provide a better metric for bonus damage, although getting damage preview to work might be problematic.

**Comment by allenhu0912** [comment]
I personally rather like the fact that you don't have to (or can't) set up a goofy back-and-forth movement path to maximize fleche damage. If fleche damage was actually tied to distance you may as well tie it to mobility instead.

**Comment by Iridar** [comment]
Not necessarily, running is not always possible, such as in very tight quarters or under overwatch/suppression. Yeah having to always waypoint around to maximize fleche damage could get a bit cheesy, but it just doesn't make sense to me while running completely different distances can result in the same damage.

**Comment by copyrite** [comment]
Small clarification: The tile from which the bonus damage is calculated from is the one at the start of the turn. This becomes relevant especially in Reaper chains.

The helpful folks on the Discord server told me that not doing `MovementTiles.Length` was a deliberate choice. That said, I would be curious what it'd do to damage arithmetic if the bonus damage were modified to be based on the shortest distance between the start and end tile of the Fleche.
